src/inet/InetInterface.h

```diff
@@ -299,6 +311,22 @@ class InterfaceIterator
      */
     bool HasBroadcastAddress();
 
+#if CHIP_SYSTEM_CONFIG_USE_ZEPHYR_NET_IF
```

Comment:
Could you make this unconditional and add at least stubs for other builds, or preferably implementations? I believe `GetHardwareAddress` for LwIP can simply use `mCurNetif->hwaddr` and `->hwaddr_len`, and for sockets `ioctl(SIOCGIFHWADDR)` similar to the socket case's `InterfaceIterator::GetFlags()`. Also, please add a trivial unit test like the one for `GetInterfaceName()` (in `TestInetInterface` in `src/inet/tests/TestInetEndPoint.cpp`) so that at least asan can catch a badly broken implementation.

Reply:
@kpschoedel ok, I tried to address it in the most current commit, please check if you are ok with that.